documentation/db-clustering-guide.html

```diff
@@ -0,0 +1,275 @@
+        </p>
+
+    </section>
+
+    <section id="connecting">
+
+        <h2>Connecting Openfire to the database cluster</h2>
+
+        <p>
+            The <a href="database.html">Database Installation Guide</a> describes how Openfire is configured to connect
+            to a database. Primarily, a <abbr title="Java DataBase Connectivity">JDBC</abbr> URL is used to define
+            most of the configuration. This URL varies depending on your database vendor. These are some examples:
+        </p>
+        <dl>
+            <dt>MySQL</dt><dd><code>jdbc:mysql://HOSTNAME:3306/DATABASENAME</code></dd>
+            <dt>Oracle</dt><dd><code>jdbc:oracle:thin:@HOSTNAME:1521:SID</code></dd>
+            <dt>Microsoft SQL Server</dt><dd><code>jdbc:sqlserver://HOSTNAME:1433;databaseName=DATABASENAME;applicationName=Openfire</code></dd>
+        </dl>
+        <p>
+            Many (but not all!) solutions for database clustering involve exposing the database cluster on more than one
+            database server address. Often, each database server in the cluster is uniquely addressable. Many database
+            vendors allow you to supply more than one server address in the JDBC URL.
+        </p>
+        <p>
+            As shown above, JDBC URLs for non-clustered database connections differ greatly depending on the database
+            vendor. The addition of cluster-specific configuration such as the definition of multiple hosts, or the
+            desired replication type, adds even more variation. When you want to define the JDBC URL that is used to
+            configure Openfire for database clustering, you will need to refer to the vendor-provided documentation.
+            Some example JDBC URLs that define more than one database server, as well as additional configuration:
+        </p>
+        <dl>
+            <dt>MySQL</dt><dd><code>jdbc:mysql://PRIMARYHOSTNAME:3306,SECONDARYHOST:3306/DATABASENAME?failOverReadOnly=false</code></dd>
+            <dt>Oracle</dt><dd><span style="color: red;">TBD</span></dd>
+            <dt>Microsoft SQL Server</dt><dd><span style="color: red;">TBD</span></dd>
+        </dl>
+
+        <section>
+
+            <h4>Links to vendor-specific documentation</h4>
+
+            <ul>
+                <li><a href="https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-multi-host-connections.html">MySQL Connector/J 8.0 Developer Guide: Multi-Host Connections</a></li>
+                <li><a href="https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/application-continuity.html#GUID-9298BB8C-0E93-44BF-BA26-A615940C0C84">Oracle Database
+                    JDBC Developer's Guide and Reference: About Configuring Oracle JDBC for Application Continuity for Java</a></li>
+                <li><span style="color: red;">TBD: complete this for all (?) supported vendors.</span></li>
+            </ul>
+
```

Comment:
Add PostgreSQL docs link

```suggestion
            <ul>
                <li><a href="https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-multi-host-connections.html">MySQL Connector/J 8.0 Developer Guide: Multi-Host Connections</a></li>
                <li><a href="https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/application-continuity.html#GUID-9298BB8C-0E93-44BF-BA26-A615940C0C84">Oracle Database
                    JDBC Developer's Guide and Reference: About Configuring Oracle JDBC for Application Continuity for Java</a></li>
                <li><a href="https://jdbc.postgresql.org/documentation/use/#connection-fail-over">PostgreSQL JDBC: Connection failover</a></li>  
                <li><span style="color: red;">TBD: complete this for all (?) supported vendors.</span></li>
            </ul>

```